Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -522,6 +584,38 @@ public:
                     // shuffle
                     ShuffleFisherYates(indices_1, cell_start_1, cell_stop_1, engine);
                     ShuffleFisherYates(indices_2, cell_start_2, cell_stop_2, engine);
+                }
+            );
+
+            // Loop over independent particle pairs
```

Comment:
```suggestion
            // Loop over independent particle pairs
            // To speed up binary collisions on GPU, we try to expose as much parallelism 
            // as possible (while avoiding race conditions): Instead of looping with one GPU 
            // thread per cell, we loop with one GPU thread per "independent pairs" (i.e. pairs 
            // that do not touch the same macroparticles, so that there is no race condition), 
            // where the number of independent pairs is determined by the lower number of 
            // macroparticles of either species, within each cell.
```